pkg/deploy/spindeploy/transformer/patches.go

```diff
@@ -0,0 +1,204 @@
+			}
+			gen.Config[k] = s
+		}
+	}
+	return nil
+}
+
+func (p *patchTransformer) applyKustomization(spinsvc string, kustomization interfaces.ServiceKustomization, config *generated.ServiceConfig) error {
+	if kustomization.Deployment != nil {
+		if config.Deployment == nil {
+			return fmt.Errorf("deployment not generated for service %s, unable to apply kustomization", spinsvc)
+		}
+		data, err := p.applyPatches(spinsvc, kustomization.Deployment, config.Deployment)
+		if err != nil {
+			return err
+		}
+		dep, err := p.asDeployment(data)
+		if err != nil {
+			return err
+		}
+		config.Deployment = dep
+	}
+	if kustomization.Service != nil {
+		if config.Service == nil {
+			return fmt.Errorf("service not generated for service %s, unable to apply kustomization", spinsvc)
+		}
+		data, err := p.applyPatches(spinsvc, kustomization.Service, config.Service)
+		if err != nil {
+			return err
+		}
+		svc, err := p.asService(data)
+		if err != nil {
+			return err
+		}
+		config.Service = svc
+	}
+	return nil
+}
+
+func (p *patchTransformer) applyPatches(spinsvc string, kustomization *interfaces.Kustomization, obj runtime.Object) ([]byte, error) {
+	originalObjJS, err := runtime.Encode(unstructured.UnstructuredJSONScheme, obj)
+	if err != nil {
+		return nil, err
+	}
+	c := &creator{}
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	for i, patch := range kustomization.Patches {
+		originalObjJS, err = getPatchedJSON(types.MergePatchType, originalObjJS, []byte(patch), gvk, c)
+		if err != nil {
+			return nil, fmt.Errorf("unable to apply json patch for %s at index %d: %v", spinsvc, i, err)
+		}
+	}
+	if kustomization.PatchesJson6902 != "" {
+		originalObjJS, err = getPatchedJSON(types.JSONPatchType, originalObjJS, []byte(kustomization.PatchesJson6902), gvk, c)
+		if err != nil {
+			return nil, fmt.Errorf("unable to apply json6902 patch for %s: %v", spinsvc, err)
+		}
+	}
+	for i, patch := range kustomization.PatchesStrategicMerge {
+		originalObjJS, err = getPatchedJSON(types.StrategicMergePatchType, originalObjJS, []byte(patch), gvk, c)
+		if err != nil {
+			return nil, fmt.Errorf("unable to apply strategic merge patch for %s at index %d: %v", spinsvc, i, err)
+		}
+	}
+	return originalObjJS, nil
+}
+
+func (p *patchTransformer) asDeployment(data []byte) (*appsv1.Deployment, error) {
+	dser := scheme.Codecs.UniversalDecoder()
+	dep := &appsv1.Deployment{}
+	obj, _, err := dser.Decode(data, &schema.GroupVersionKind{
+		Group:   "apps",
+		Version: "v1",
+		Kind:    "Deployment",
+	}, dep)
+	if err != nil {
+		return nil, err
+	}
+	if dep, ok := obj.(*appsv1.Deployment); ok {
+		return dep, nil
+	}
+	return nil, errors.New("not parsed as a deployment, patching failed")
+}
+
+func (p *patchTransformer) asService(data []byte) (*v1.Service, error) {
+	dser := scheme.Codecs.UniversalDecoder()
+	svc := &v1.Service{}
+	obj, _, err := dser.Decode(data, &schema.GroupVersionKind{
+		Group:   "apps",
+		Version: "v1",
+		Kind:    "Deployment",
```

Comment:
Isn't this gvk for service `v1.Service` as is line 162?

Reply:
Yes but it still works. Makes me wonder how...